crates/ruma-api-macros/src/api/request/outgoing.rs

```diff
@@ -226,11 +226,9 @@ impl Request {
                             "application/json",
                         );
 
-                    let mut req_headers = req_builder
-                        .headers_mut()
-                        .expect("`http::RequestBuilder` is in unusable state");
```

Comment:
I thought about using `ok_or` and return an `IntoHttpError`, but it would need a new variant. Also, It seems that the specific error can't be extracted from the builder, so it would need to be a "generic" variant like "InvalidRequest" or similar.